lib/parser.js

```diff
@@ -98,16 +101,18 @@ function expectAnyToken(expectedTokens, tokens)
 
   if(!foundToken)
   {
-    throw new Error(`Invalid parse state! Expected any of: '${expectedTokens}' but got: '${tokens[0]}' from chain: [${tokens}]`);
+    throw new Error(`Invalid parse state! Expected any of: '${expectedTokens}' but got: '${tokens[0]}' from chain: [${tokens}]. ${parseInfo(parseContext)}`);
   }
 }
 
 /**
  * Joins the set of tokens into a string such that joinTokens([a, b]) returns "a b".
```

Comment:
This comment is no longer exactly true after this change.  Does joinTokens need to take a parseContext anyway?